src/workerd/io/trace-stream.c++

```diff
@@ -971,6 +971,8 @@ kj::Promise<WorkerInterface::CustomEvent::Result> TailStreamCustomEventImpl::sen
     rpc::EventDispatcher::Client dispatcher) {
   auto revokePaf = kj::newPromiseAndFulfiller<void>();
 
+  // TODO(streaming-tail): Session is currently being reported as canceled when using
```

Comment:
The tail stream gets reported as canceled when running the downstream STW test via RPC – this code is largely copied from JSRPC, unfortunately I do not understand it well enough to investigate this properly. Note that this is an entirely different stream than the TailStreamTarget stream where previous issues have been fixed, even though the error message is identical. Tests are still passing, so I'm just adding a TODO here for now.

Reply:
Likely the waitUntilTasks